openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/authorities/CaseAuthoritiesWebScript.java

```diff
@@ -0,0 +1,174 @@
+import org.alfresco.service.cmr.repository.InvalidNodeRefException;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.security.AuthenticationService;
+import org.alfresco.service.cmr.security.AuthorityService;
+import org.alfresco.service.cmr.security.AuthorityType;
+import org.alfresco.service.cmr.security.PersonService;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.util.Pair;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.extensions.webscripts.WebScriptException;
+import org.springframework.extensions.webscripts.WebScriptRequest;
+import org.springframework.extensions.webscripts.WebScriptResponse;
+import org.springframework.stereotype.Component;
+
+@Component
+@WebScript(description = "Get groups/users for case by case type", families = {"Authorities"})
+public class CaseAuthoritiesWebScript {
+
+    private static final String CREATED_ON_OPEN_E = "OPENE";
+    private static final int MAX_ITEMS = 10000;
+    private static final PagingRequest PAGING_REQUEST = new PagingRequest(MAX_ITEMS);
+    private static final List SORT_PROPS = Arrays.asList(
+            new Pair(ContentModel.PROP_FIRSTNAME, true),
+            new Pair(ContentModel.PROP_LASTNAME, true)
+    );
+    private static final List<QName> FILTER_PROPS = Arrays.asList(
+            ContentModel.PROP_FIRSTNAME,
+            ContentModel.PROP_LASTNAME);
+
+    @Autowired
+    private AuthorityService authorityService;
+    @Autowired
+    private NodeService nodeService;
+    @Autowired
+    private AuthenticationService authenticationService;
+    @Autowired
+    private PersonService personService;
+
+    @Uri(value = "/api/openesdh/{caseType}/authorities", method = HttpMethod.GET, defaultFormat = "json")
+    public Resolution getCaseAuthorities(WebScriptRequest req, WebScriptResponse res) throws JSONException {
+        String caseType = getRequiredTemplateParam(req, "caseType");
+        //groups for case type
+        List<JSONObject> jsonAuthorities = getAuthoritiesByType(req, caseType);
+        //other groups created in openE
+        jsonAuthorities.addAll(getAuthoritiesByType(req, null));
+        //users
+        jsonAuthorities.addAll(getPeople(req));
+        return WebScriptUtils.jsonResolution(
+                new JSONObject().put("data",
+                        new JSONObject().put("items",
+                                new JSONArray(jsonAuthorities))));
+    }
+
+    private List<JSONObject> getAuthoritiesByType(WebScriptRequest req, String caseType) throws JSONException {
+        PagingResults<AuthorityInfo> authorities = authorityService.getAuthoritiesInfo(AuthorityType.GROUP,
+                null,//zone
+                req.getParameter("filter"),
```

Comment:
The @RequestParam annotation for the "filter" perhaps would be more appropriate.